Civi/Test/PageWrapper.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Civi\Test;
+
+class PageWrapper {
+
+  /**
+   * @var \CRM_Core_Page
+   */
+  private \CRM_Core_Page $page;
+
+  private array $statusMessage;
+
+  /**
+   * @var false|string
+   */
+  private string $output;
+
+  /**
+   * @param string $class
+   * @param array $urlParameters
+   */
+  public function __construct(string $class, array $urlParameters = []) {
+    $_GET = $urlParameters;
+    $this->page = new $class();
+    $_SERVER['REQUEST_METHOD'] = 'GET';
+    $_REQUEST = array_merge($_REQUEST, $urlParameters);
```

Comment:
@eileenmcnaughton  Have you seen `LocalHttpTest`? It's basically the same idea -- but more developed. There are bunch of usage-examples in [OembedTest](https://github.com/civicrm/civicrm-core/blob/5.77/ext/oembed/tests/phpunit/CRM/Oembed/Page/OembedTest.php#L26) (*search for `$this->client`*) and a few other places (like [testHttpUnsubscribe](https://github.com/civicrm/civicrm-core/blob/5f254d8b28d57d404b489004133f8ff16ac1fa25/tests/phpunit/CRM/Mailing/MailingSystemTestBase.php#L100)).